pkg/config/app/admin.go

```
@@ -50,16 +50,26 @@ var _ = &AdminConfig{}
 
 func (c *AdminConfig) Sanitize() {
 	c.Engine.Sanitize()
-	c.Discovery.Sanitize()
+	for _, d := range c.Discovery {
+		d.Sanitize()
+	}
 	c.Store.Sanitize()
 	c.Console.Sanitize()
 	c.Diagnostics.Sanitize()
 }
 
 func (c *AdminConfig) PostProcess() error {
+	discoveryPostProcess := func() error {
+		for _, d := range c.Discovery {
+			if err := d.PostProcess(); err != nil {
+				return err
+			}
+		}
+		return nil
+	}
 	return multierr.Combine(
 		c.Engine.PostProcess(),
-		c.Discovery.PostProcess(),
+		discoveryPostProcess(),
```

If a discovery throw exception, ignore whether it meets expectations here.

---

It' designed to fail fast, because if a config of dsicovery is invalid, the start and run process would return err either.